guides/ci/travis.md

```diff
@@ -0,0 +1,364 @@
+
+```yml
+# .travis.yml
+language: bash
+
+services:
+  - docker
+
+script:
+  - docker run -v $PWD:/src -w /src crystallang/crystal:0.31.1 crystal spec spec/game_of_life_spec.cr
+```
+
+**Note:** We may find the list with the different official [Crystal docker images](https://hub.docker.com/r/crystallang/crystal/tags) at [DockerHub](https://hub.docker.com/r/crystallang/crystal)
+
+**Note:** This is a great trick that will let us use different Crystal versions without the need of actually installing the compiler and without the need of creating a `Dockerfile` configuration file.
+
+### Using `latest`, `nightly` and a specific Crystal release all together!
+
+For using 3 versions of the compiler we are going to use a different configuration file based on [Build Matrix](https://docs.travis-ci.com/user/customizing-the-build#build-matrix)
+Here is the example:
+
+```yaml
+# .travis.yml
+matrix:
+ include:
+   - language: crystal
+     crystal:
+       - latest
+     script:
+       - crystal spec
+
+   - language: crystal
+     crystal:
+       - nightly
+     script:
+       - crystal spec
+
+   - language: bash
+     services:
+       - docker
+     script:
+       - docker run -v $PWD:/src -w /src crystallang/crystal:0.31.1 crystal spec spec/game_of_life_spec.cr
+```
+
+## Using external dependencies
+
+Before presenting some examples, it is important to mention that there are many ways to achieve this and **it will heavily depend on our development workflow**.
+
+With this in mind, here is a first example installing Git using Travis CI configuration file:
+
+```yaml
+# .travis.yml
+language: crystal
+crystal:
+  - latest
+
+before_install:
+  - sudo apt-get -y install git
+
+addons:
+  apt:
+    update: true
+
+script:
+  - git --version
+  - crystal spec spec/game_of_life_spec.cr
+```
+
+**Note:** We are using `git` just for printing GIT’s version, as an example of external dependency.
+
+**Note:** Git is already installed by TravisCI and available but (again) we are using it as an example of installing and using a dependency.
+
+The same can be accomplished using a containerization service like Docker. Here is an example of a `Dockerfile` installing Git:
+
+```dockerfile
+# Dockerfile
+FROM crystallang/crystal:latest
+
+RUN apt-get update && apt-get install -y git
+```
+
+And here is the TravisCI configuration file:
+
+```yml
+# .travis.yml
+language: bash
+
+services:
+  - docker
+
+before_install:
+  # build image using Dockerfile:
+  - docker build -t testing .
+  # Git installed by Travis CI
+  - git --version # => 2.21.0
+
+script:
+  # run specs in the container
+  - docker run -v $PWD:/src -w /src testing crystal spec spec/game_of_life_spec.cr
+  # Git installed in Docker container (see Dockerfile)
+  - docker run -v $PWD:/src -w /src testing git --version # => 2.17.1
+```
+
+We may notice that GIT’s version running in the `testing` container is different from Git running by default in Travis CI.
```

Comment:
I would focus on the fact that the external dependency needs to be in the docker images.